rust/src/mysql/mysql.rs

```diff
@@ -0,0 +1,1133 @@
+            || self.state_progress == StmtPrepareResponseReceived
+            || self.state_progress == StmtExecResponseReceived
+    }
+
+    pub fn parse_response(&mut self, flow: *const Flow, i: &[u8]) -> AppLayerResult {
+        // We're not interested in empty responses.
+        if i.is_empty() {
+            return AppLayerResult::ok();
+        }
+
+        if self.response_gap {
+            if !probe(i) {
+                SCLogDebug!("Suricata interprets there's a gap in the response");
+                return AppLayerResult::ok();
+            }
+
+            // It seems we're in sync with a message header, clear gap state and keep parsing.
+            self.response_gap = false;
+        }
+
+        let mut start = i;
+
+        while !start.is_empty() {
+            if self.state_progress == MysqlStateProgress::Finished || self.invalid_state_resp() {
+                return AppLayerResult::ok();
+            }
+            match MysqlState::state_based_resp_parsing(
+                self.state_progress,
+                start,
+                self.client_flags,
+            ) {
+                Ok((rem, response)) => {
+                    start = rem;
+
+                    SCLogDebug!("Response is {:?}", &response);
+                    if let Some(state) = self.response_next_state(response, flow) {
+                        self.state_progress = state;
+                    }
+                    let tx_completed = self.is_tx_completed();
+                    if let Some(tx) = self.find_or_create_tx() {
+                        if tx_completed {
+                            tx.complete = true;
+                        }
+                    }
+                }
+                Err(nom7::Err::Incomplete(_needed)) => {
+                    let consumed = i.len() - start.len();
+                    let needed_estimation = start.len() + 1;
+                    SCLogDebug!(
+                        "Needed: {:?}, estimated needed: {:?}, start is {:?}",
+                        _needed,
+                        needed_estimation,
+                        &start
+                    );
+                    return AppLayerResult::incomplete(consumed as u32, needed_estimation as u32);
+                }
+                Err(err) => {
+                    SCLogError!(
+                        "Error while parsing MySQL response, state: {:?} err: {:?}",
+                        self.state_progress,
+                        err,
+                    );
+                    return AppLayerResult::err();
+                }
+            }
+        }
+
+        // All Input was fully consumed.
+        AppLayerResult::ok()
+    }
+
+    pub fn on_request_gap(&mut self, _size: u32) {
+        self.request_gap = true;
+    }
+
+    pub fn on_response_gap(&mut self, _size: u32) {
+        self.response_gap = true;
+    }
+}
+
+/// Probe for a valid mysql message
+pub fn probe(input: &[u8]) -> bool {
+    if parse_packet_header(input).is_ok() {
```

Comment:
Can not it say "not enough data to know" ?

Reply:
Looks like `parse_packet_header` accepts pretty much everything

Reply:
```rust
/// Probe for a valid mysql message
pub fn probe(i: &[u8]) -> IResult<&[u8], ()> {
    let (i, _) = parse_packet_header(i)?;
    Ok((i, ()))
}
```
Did you mean this?

Reply:
No, I mean that the probing parser is likely not strict enough

Reply:
I have another probe function `rs_mysql_probing_tc` and `rs_mysql_probing_ts` for real probing